Fix manager docker version issue with CI/CD deployment in custom projects

richturner opened the pull request:
Fixes #1666

wborn commented on .github/workflows/ci_cd.yml:
```diff
@@ -213,32 +210,28 @@ jobs:
+              exit 1
             fi
+
+            echo "ref=$version" >> $GITHUB_OUTPUT
```
The master value was used so you can more easily use/test snapshot versions in a custom project. That's no longer possible with these changes.

richturner replied:
We don't build snapshot versions of docker containers

wborn commented on .github/workflows/ci_cd.yml:
```diff
@@ -415,13 +322,18 @@ jobs:
                  npmPublishTag = eventConfig['distribute']['npm']
 
-          if isMainRepo and ((eventName == "release" and isRelease) or (eventName != "release" and not isRelease)):
+          if isMainRepo and (isRelease or (eventName != "release" and not isRelease)):
```
This construct was there so it only tries to release on tags and not on the master branch. It could try to re-release for instance if you deploy a test instance just after creating a release. If there are no commits after just creating a release the release plugin resolves the version to a release and not a snapshot version.

richturner replied:
Thanks for the heads up.

The interaction with the versioning plugin and github actions seems a little confusing, I'll reinstate this for now.

richturner commented:
@wborn some info:

* Removed support for `managerDockerPush` in `ci_cd.json` which was intended to auto update custom projects when ta new manager docker image was available, wasn't used and doesn't make much sense now submodule is removed
* Added `CLEAN_INSTALL=true` to env for demo and staging.demo deployment (this was missing)

wborn commented:
@richturner the CI/CD workflow now fails with this changes, see: https://github.com/openremote/openremote/actions/runs/13367671078/job/37328995588
